### DISCUSSION TITLE
How to use PDM mic properly?

### DISCUSSION THREAD

**Discussion by yujonglee** [pschatzmann/arduino-audio-tools#919] [Q&A] [answered]
I saw in https://github.com/pschatzmann/arduino-audio-tools/commit/2174cecef7f0341d32e5ebbe30919e8cd6e7758f , example about PDM mic is added.

I have some question:

1. What is `cfg.signal_type = PDM` for? I searched the codebase but `signal_type` seems to do nothing.
2. https://github.com/pschatzmann/arduino-audio-tools/blob/33018b177ed6eea75ebb46c1388f82a802125d8d/examples/examples-stream/streams-i2s_pdm-serial/streams-i2s_pdm-serial.ino#L29 Can you explain what does `not used` mean here?
 In the [ESP doc](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/i2s.html#introduction) said I need `CLK` and `DATA`, and I thought `.pin_bck` is corresponding to `CLK`. 
3. If I use two PDM mic connected, (select pin of one is high and other is low) what I2S format should I use instead of `I2S_CHANNEL_FMT_ALL_RIGHT` in example? 
```c
enum I2SFormat {
  I2S_STD_FORMAT,
  I2S_LSB_FORMAT,
  I2S_MSB_FORMAT,
  I2S_PHILIPS_FORMAT,
  I2S_RIGHT_JUSTIFIED_FORMAT,
  I2S_LEFT_JUSTIFIED_FORMAT,
  I2S_PCM,
};
```

Thank you.

**Comment by pschatzmann** [comment, marked as answer]
I suggest that you read the [documentation](https://github.com/pschatzmann/arduino-audio-tools/wiki/External-ADC#mp34dt01---pdm-digital-mems-microphone) or just use the corresponding example

**Reply by yujonglee** [reply]
@pschatzmann Thanks. I got it. `CLK` is corresponding to `pin_ws`.

Just one thing: How about `I2SFormat` that I mentioned in `3`?

https://github.com/pschatzmann/arduino-audio-tools/blob/33018b177ed6eea75ebb46c1388f82a802125d8d/src/AudioI2S/I2SESP32-New.h#L113

https://github.com/pschatzmann/arduino-audio-tools/blob/33018b177ed6eea75ebb46c1388f82a802125d8d/src/AudioI2S/I2SESP32-New.h#L202

Seems like when using `PDM`, `I2SFormat` does not matter but just wanna classify.

Thank you.

**Comment by pschatzmann** [comment]
That would be my understanding as well: just notice that the new Driver is only available when you use the latest IDF versions.
In Arduino the old I2S API is used...
